implementation/service_discovery/src/service_discovery_impl.cpp

```diff
@@ -2355,14 +2357,14 @@ service_discovery_impl::handle_eventgroup_subscription(
         boost::system::error_code ec;
         // TODO: Add session id
         VSOMEIP_ERROR << __func__
-                << ": Requested major version:[" << (uint32_t) _major
+                << ": Requested major version:[" << +static_cast<uint32_t>(_major)
                 << "] in subscription to service: ["
                 << std::hex << std::setfill('0')
                 << std::setw(4) << _service << "."
                 << std::setw(4) << _instance << "."
                 << std::setw(4) << _eventgroup << "]"
                 << " does not match with services major version:["
-                << (uint32_t) _info->get_major() << "] subscriber: "
+                << +static_cast<uint32_t>(_info->get_major()) << "] subscriber: "
```

Comment:
That one's not a typo, using the `+` promotes the value to an int for display purposes.  I suppose in this case I could probably drop the `static_cast`.

Without the plus though, often a bland ascii character is output instead of the value.

For reference:
- [SO answer where I found this](https://stackoverflow.com/a/31991844/1861346)
- [Integer Promotion](https://stackoverflow.com/a/31991844/1861346)

Reply:
According to your reference, both `static_cast<uint32_t>(_info->get_major())` and `+_info->get_major()` do the job of promoting the value to an int so that it can be displayed correctly.

If that is the case, I would prefer if you used the more explicit option to promote value, which is the `static_cast`.